webapp/src/components/task_creation_form/containers/FileContainer.vue

```diff
@@ -93,14 +118,48 @@ const emit = defineEmits<Emits>()
 
 const fileName = ref('')
 const upload = ref<HTMLInputElement>()
+  const toast = useToast()
 
 function onChange(): void {
+  showUploadingToast()
   if (upload.value === undefined) return
   const files = upload.value.files
   if (files === null) return
 
   // fill in the vee-field to trigger yup validation
   fileName.value = (files.length > 0) ? files[0].name : ''
   emit('input', files)
+  toast.success("Upload complete")
+}
+// function onDragOver(event: DragEvent): void {
+//   // Optional: Add visual feedback
+// }
+
+function onDrop(event: DragEvent): void {
+  if (!available) return;
+  showUploadingToast()
+
+  const files = event.dataTransfer?.files;
+  if (files && files.length > 0) {
+    // Process the files
+    fileName.value = files[0].name;
+    toast.success("Upload complete")
+    emit('input', files);
+  } else {
+    toast.error("Upload failed, please try again")
+  }
+}
+
+function showUploadingToast() {
+  toast.info("Uploading your data in the browser (your data stays on your computer)")
+}
+
+// clearInput method
```

Comment:
This comment doesn't provide mumch information you can remove it or add more information